### ISSUE TITLE
Improve formation rank custom formations.

### ISSUE THREAD

**Issue by GoogleFrog** [ZeroK-RTS/Zero-K#4961]
There are two issues with formation rank at the moment.
- [ ] Tangent detection is too jagged. This causes the preview dots to swing around as they pass over small bumps, and a command issued with the janky preview will end up with a column of units standing at an angle.
- [ ] Orienting the formation based on which direction the line was drawn is bad. People do not naturally draw things either clockwise or anticlockwise. Muscle memory for similar UIs (like drag selecting) is not set up for it. I don't know the solution, but it should be solved.

Here are two approaches for the second issue.
- The simple one is to anchor the orientation of the line at the average position of the units receiving the command. This isn't super simple as lines can be drawn as loops or non-orientable squiggles. We would just have to accept that people who draw ambiguous formations are going to get random results. At least the preview makes this predictable, if not controllable.
- An improvement the above would probably require a bunch of heuristics. Perhaps take the start of the line being drawn and compare it to the average of the units.
- A less simple approach would be to do whatever reduces the distance between units and their destination. This sounds expensive.

The best solution probably involves the first approach with a bunch of heuristics and fallbacks for various edge cases.

**Comment by sprunk** [comment]
I don't think this is fully solveable because sometimes the same line can mean two things. Consider two situations:
![image](https://github.com/ZeroK-RTS/Zero-K/assets/2573076/2adba6a9-bb1a-48d5-8cc2-ba26c7da0b05)
I'm getting pushed and I want to retreat a bit. The formation should stay the same, i.e. riots are north of the arty.

![image](https://github.com/ZeroK-RTS/Zero-K/assets/2573076/b05f17bc-b839-42d0-a733-58ca2fdc54f7)
I got flanked and I want to reposition. The formation should reverse, the riots should move south of the arty.

Drag direction solves this and is also how it works in other games where facing matters (for example Total War series).
